whois-nrtm4/src/main/java/net/ripe/db/nrtm4/SnapshotFileGenerator.java

```diff
@@ -0,0 +1,99 @@
+package net.ripe.db.nrtm4;
+
+import com.google.common.base.Stopwatch;
+import net.ripe.db.nrtm4.dao.NrtmDocumentType;
+import net.ripe.db.nrtm4.dao.NrtmSource;
+import net.ripe.db.nrtm4.dao.NrtmVersionInfo;
+import net.ripe.db.nrtm4.dao.NrtmVersionInfoRepository;
+import net.ripe.db.nrtm4.dao.SnapshotFile;
+import net.ripe.db.nrtm4.dao.SnapshotFileRepository;
+import net.ripe.db.nrtm4.domain.PublishableSnapshotFile;
+import net.ripe.db.nrtm4.domain.SnapshotFileSerializer;
+import net.ripe.db.nrtm4.util.NrtmFileUtil;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.text.DecimalFormat;
+import java.util.Optional;
+
+
+@Service
+public class SnapshotFileGenerator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SnapshotFileGenerator.class);
+
+    private final NrtmVersionInfoRepository nrtmVersionInfoRepository;
+    private final SnapshotFileSerializer snapshotFileSerializer;
+    private final SnapshotFileRepository snapshotFileRepository;
+    private final SnapshotObjectSynchronizer snapshotObjectSynchronizer;
+    private final NrtmFileStore nrtmFileStore;
+
+    public SnapshotFileGenerator(
+        final NrtmVersionInfoRepository nrtmVersionInfoRepository,
+        final SnapshotFileSerializer snapshotFileSerializer,
+        final SnapshotFileRepository snapshotFileRepository,
+        final SnapshotObjectSynchronizer snapshotObjectSynchronizer,
+        final NrtmFileStore nrtmFileStore
+    ) {
+        this.nrtmVersionInfoRepository = nrtmVersionInfoRepository;
+        this.snapshotFileSerializer = snapshotFileSerializer;
+        this.snapshotFileRepository = snapshotFileRepository;
+        this.snapshotObjectSynchronizer = snapshotObjectSynchronizer;
+        this.nrtmFileStore = nrtmFileStore;
+    }
+
+    public Optional<PublishableSnapshotFile> createSnapshot(final NrtmSource source) {
+        final String method = "createSnapshot";
+        // Get last version from database.
+        final Optional<NrtmVersionInfo> lastVersion = nrtmVersionInfoRepository.findLastVersion(source);
+        NrtmVersionInfo version;
+        LOGGER.info("{} lastVersion.isEmpty() {}", method, lastVersion.isEmpty());
+        if (lastVersion.isEmpty()) {
+            version = snapshotObjectSynchronizer.initializeSnapshotObjects(source);
```

Comment:
can we directly call initializeSnapshotObjects through jmx call? It will separate  the actual genration of delata and subsequent snapshot from initial generation , somethiung like we do in GrsImporterJmx